javascript/phone-number/phone-number.js

```diff
@@ -0,0 +1,39 @@
+var PhoneNumber = function(input){
+	//validate the input type
+	if(typeof input !== "string") {
+		throw "Invalid input type. Please give a number in string and try again.";
+	}
+	this.num = input;
+};
+
+PhoneNumber.prototype.number = function(){
+	//clear the non-number char in the string
+	var regex = /[0-9]/g;
+	var n = this.num.match(regex).join("");
+
+	var len = n.length;
+	if(len < 10 || len > 11) { return "0000000000"; }
+	if(len === 10) { return n; }
+	if(len === 11) {
+		if(n.charAt(0) === "1") {
+			return n.substring(1);
+		} else {
+			return "0000000000";
+		}
+	} 
```

Comment:
This set of conditionals is bit more complex than required for the tests. You may could improve it by:

- DRY up your `0000000000` string into a `var`.
- Eliminate the nesting and reduce the number of conditionals by having your method always return `n` with your conditional modifying `n` when necessary.